src/deploymentTemplate.ts

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as utilities from './utilities';
+import { Constants } from './constants';
+import { TelemetryClient } from './telemetryClient';
+
+var request = require('request');
+
+export class DeploymentTemplate {
+    constructor() {}
+
+    public displayDeploymentTemplateMenu() {
+
+        TelemetryClient.sendEvent('deployment');
+        // currently only one option is available, so first menu won't be displayed yet
+        this.selectQuickstartTemplate();
+    }
+
+    public selectQuickstartTemplate() {
+
+        // get list of directories from here:
+        // https://api.github.com/[repo_user/repo_name]/contents/
+        let repo: string = utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo);
+
+        var http = require('https');
+        let __this = this;
+
+            http.get({
+                host: 'api.github.com',
+                path: '/repos/' + repo + '/contents/',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+            }, function(response) {
+                // Continuously update stream with data
+                var body = '';
+                response.on('data', function(d) {
+                    body += d;
+                });
+                response.on('end', function() {
+        
+                    // Data reception is done, do whatever with it!
+                    var parsed = JSON.parse(body);
+                    let items: vscode.QuickPickItem[] = [];
+            
+                    for (var i in parsed)
+                    {
+                        // list only directories and skip known directories that don't contain templates
+                        if (parsed[i].type == "dir" && parsed[i].name != "1-CONTRIBUTION-GUIDE" && parsed[i].name != ".github") {
+                            items.push({ label: parsed[i].name, description: null });
+                        }
+
+                    }
+                    vscode.window.showQuickPick(items).then(selection => {
+                        // the user canceled the selection
+                        if (!selection) {
+                          return;
+                        }        
+                        
+                        __this.retrieveTemplate(selection.label);
+                    });
+                });
+            });
+        
+    }
+
+    public retrieveTemplate(templateName: string) {
+        var http = require('https');
+        let __this = this;
+        let repo: string = utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo);
```

Comment:
will support customize github repo later? if yes, please don't expose the configuration Config_deploymentTemplatesGitHubRepo for now, but use hardcode url.

Reply:
disabled for now

Reply:
utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo); always null? since the config item is removed